packages/@react-aria/grid/src/useGrid.ts

```diff
@@ -77,7 +83,65 @@ export function useGrid<T>(props: GridProps, state: GridState<T, GridCollection<
     gridProps['aria-colcount'] = state.collection.columnCount;
   }
 
+  // Many screen readers do not announce when items in a grid are selected/deselected.
+  // We do this using an ARIA live region.
+  let selection = state.selectionManager.rawSelection;
+  let lastSelection = useRef(selection);
+  useUpdateEffect(() => {
+    if (!state.selectionManager.isFocused) {
+      return;
+    }
+
+    let addedKeys = diffSelection(selection, lastSelection.current);
+    let removedKeys = diffSelection(lastSelection.current, selection);
+
+    // If adding or removing a single row from the selection, announce the name of that item.
+    let messages = [];
+    if (addedKeys.size === 1 && removedKeys.size === 0) {
+      let addedText = getRowText(addedKeys.keys().next().value);
+      if (addedText) {
+        messages.push(formatMessage('selectedItem', {item: addedText}));
+      }
+    } else if (removedKeys.size === 1 && addedKeys.size === 0) {
+      let removedText = getRowText(removedKeys.keys().next().value);
+      if (removedText) {
+        messages.push(formatMessage('deselectedItem', {item: removedText}));
+      }
+    }
+
+    // Announce how many items are selected, except when selecting the first item.
+    if (state.selectionManager.selectionMode === 'multiple') {
+      if (messages.length === 0 || selection === 'all' || selection.size > 1 || lastSelection.current === 'all' || lastSelection.current.size > 1) {
+        messages.push(selection === 'all'
+          ? formatMessage('selectedAll')
+          : formatMessage('selectedCount', {count: selection.size})
+        );
+      }
+    }
+
+    if (messages.length > 0) {
+      announce(messages.join(' '));
+    }
+
+    lastSelection.current = selection;
+  }, [selection]);
+
   return {
     gridProps
   };
 }
+
+function diffSelection(a: Selection, b: Selection): Set<Key> {
+  let res = new Set<Key>();
+  if (a === 'all' || b === 'all') {
+    return res;
+  }
+
+  for (let key of a.keys()) {
+    if (!b.has(key)) {
+      res.add(key);
+    }
+  }
+
+  return res;
```

Comment:
I think it would need to be this because of function binding:

```js
return new Set([...a].filter(key => !b.has(key)));
```

Also it might be slower because of the two additional array allocations/copies. Didn't measure though. Personally, the loop is easier to read. 🤷